base/fs.cpp

```diff
@@ -210,6 +210,47 @@ std::string get_file_title_with_path(const std::string& filename)
   return filename;
 }
 
+std::string get_relative_path(const std::string& filename, const std::string& base_path)
+{
+  std::vector<std::string> baseDirs; 
+  split_string(base_path, baseDirs, "/\\");
+
+  std::vector<std::string> toParts;
+  split_string(filename, toParts, "/\\");
+
+  // Find the common prefix
+  auto itFrom = baseDirs.begin();
+  auto itTo = toParts.begin();
+
+  while (itFrom != baseDirs.end() && itTo != toParts.end() && *itFrom == *itTo) {
+    ++itFrom;
+    ++itTo;
+  }
+
+  if (itFrom == baseDirs.begin() && itTo == toParts.begin()) {
+    // No common prefix
+    return filename;
+  }
+
+  // Calculate the number of directories to go up from base path
+  std::string relativePath;
+  for (auto it = itFrom; it != baseDirs.end(); ++it) {
+      std::string upDir("..");
+      upDir.push_back(base::path_separator);
+      relativePath += upDir;
+  }
```

Comment:
Probably you can simplify this with
```suggestion
  std::string relativePath;
  for (auto it = itFrom; it != baseDirs.end(); ++it)
    relativePath = base::join_path(relativePath, "..");
```
In other case you should move upDir outside the for-loop, and fix the indentation to 2 spaces.